### PR TITLE
HIVE-26280: Copy more data into COMPLETED_COMPACTIONS for better supportability

### PR DESCRIPTION
**Pull request by klcopp**
### What changes were proposed in this pull request?
- Relevant columns in COMPACTION_QUEUE should be present in COMPLETED_COMPACTIONS, and their data should be copied to COMPLETED_COMPACTIONS. Also, make sure that SYS.COMPLETED_COMPACTIONS, SYS.COMPACTION_QUEUE, and SYS.COMPACTIONS contain all the information they should

### Does this PR introduce _any_ user-facing change?
Sys tables will have more columns

### CODE REVIEW

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/CompactionInfo.java [hunk 1]
```diff
@@ -59,6 +59,9 @@ public class CompactionInfo implements Comparable<CompactionInfo> {
   public boolean tooManyAborts = false;
   public boolean hasOldAbort = false;
   public long retryRetention = 0;
+  public long nextTxnId = 0;
```
Comment (deniskuzZ) [inline standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/CompactionInfo.java:62]: this is needed for `show compactions`, right?

File: metastore/scripts/upgrade/hive/hive-schema-4.0.0-alpha-2.hive.sql [hunk 2]
```diff
@@ -1138,6 +1144,9 @@ CREATE EXTERNAL TABLE IF NOT EXISTS `COMPLETED_COMPACTIONS` (
   `CC_HIGHEST_WRITE_ID` bigint,
   `CC_HADOOP_JOB_ID` string,
   `CC_ERROR_MESSAGE` string,
+  `CC_NEXT_TXN_ID` bigint,
```
Comment (deniskuzZ) [inline metastore/scripts/upgrade/hive/hive-schema-4.0.0-alpha-2.hive.sql:1147]: I think, CC_NEXT_TXN_ID` & `CC_TXN_ID are only relevant for in-flight processing and not useful during the troubleshooting (txnIds would be already gone)

**Review by deniskuzZ — COMMENTED** [review]
LGTM, just 1 comment if we really need to propagate CC_NEXT_TXN_ID & CC_TXN_ID

File: metastore/scripts/upgrade/hive/hive-schema-4.0.0-alpha-2.hive.sql (hunk 2 above)
Comment (klcopp) [reply metastore/scripts/upgrade/hive/hive-schema-4.0.0-alpha-2.hive.sql:1147]: Well, just thinking about troubleshooting situations... txnIds will be present in the logs and might be still left open in the TXNS table. NEXT_TXN_ID will tell you when the cleaner should have picked up the compaction for cleaning.
Do you think there's any downside to propagating them?

**Review by deniskuzZ — APPROVED** [review]
LGTM, consider the comment above, pending tests.

**Pull request merged**